src/huff0/huff0_decoder.rs

```diff
@@ -236,10 +264,13 @@ impl<'t> HuffmanDecoder<'t> {
         Ok(num_bits)
     }
 
+    /// Advance the internal cursor to the next symbol. After this, you can call `decode_symbol`
+    /// to read from the new position.
     pub fn next_state(
         &mut self,
         br: &mut BitReaderReversed<'_>,
     ) -> Result<u8, HuffmanDecoderError> {
+        // I have no clue how this works
```

Comment:
This comment should be removed ;)

Reply:
self.state stores a window of the bit stream. The table can be indexed via this state, telling you how many bits identify the current symbol. These bits that identify the current symbol get shifted out, new bits are read from the stream and appended at the end of the current stream.

Reply:
Documentation updated, probably needs a second set of eyes.

Reply:
Looks good :)